tests/fixtures/typescript-types

```diff
@@ -28,7 +28,7 @@
 {function(() => string): void}
 {X<() => string>}
 {new (x: number) => string}
-//{[it, needs, to, be]}
+{[it, needs, to, be]}
```

Comment:
can you add a few more tests? I'd like to see one per rule that got altered in the pegjs file, although that's hard to verify sometimes.

Also tests of `[]`, `[x]` and `[][]`.

Reply:
Way ahead of you:
https://github.com/EB-Forks/jsdoctypeparser/blob/9329a0bbca7eea5a78ee999e5cdd5acd3c63cd58/tests/test_parsing.js#L494-501

https://github.com/EB-Forks/jsdoctypeparser/blob/9329a0bbca7eea5a78ee999e5cdd5acd3c63cd58/tests/test_parsing.js#L503-512

https://github.com/EB-Forks/jsdoctypeparser/blob/9329a0bbca7eea5a78ee999e5cdd5acd3c63cd58/tests/test_parsing.js#L526-540